File: pythonCode/Advisor.py
from Instructor import Instructor
from ApprovalState import ApprovalState

class Advisor(Instructor):

    def __init__(self):
        pass

    def getListOfStudents(self):
        try:
            x = self.__listOfStudents
        except AttributeError:
            self.__listOfStudents = list()
        
        return self.__listOfStudents

    def getListOfApplications(self):
        try:
            x = self.__listOfApplications
        except AttributeError:
            self.__listOfApplications = list()

        return self.__listOfApplications

    def approveApplication(self, lectureRegistirationApplication, lectureSession):
        # generated source for method approveApplication 
        lectureRegistirationApplication.getSessionsSentForApproval().put(lectureSession, ApprovalState.Approved)
        lectureRegistirationApplication.getStudent().getSchedule().getListOfLectureSessions().add(lectureSession)

    def rejectApplication(self, lectureRegistirationApplication, lectureSession):
        # generated source for method rejectApplication 
        lectureRegistirationApplication.getSessionsSentForApproval().put(lectureSession, ApprovalState.Rejected)

File: Iteration 1/JavaCode/person/Advisor.java
package person;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;
import java.util.Map;

import Debt_LRA_Transcript.LectureRegistrationApplication;
import Enums.ApprovalState;
import Enums.InstructorType;
import IDs.InstructorID;
import lecture.Lecture;
import lecture.LectureSession;

//Kaan Camci 150119063
public class Advisor extends Instructor {

	private List<Student> listOfStudents;
	private List<LectureRegistrationApplication> listOfApplications;

	public List<Student> getListOfStudents() {
		return listOfStudents;
	}

	public List<LectureRegistrationApplication> getListOfApplications() {
		return listOfApplications;
	}

	public void approveApplication(LectureRegistrationApplication lectureRegistirationApplication) {
		for(LectureRegistrationApplication lra: listOfApplications) {
			Map<LectureSession, ApprovalState> lsa = lra.getSessionsSentForApproval();
			for(LectureSession ls: lsa.keySet()) {
				lsa.put(ls, ApprovalState.Approved);
			}
		}
	}

	public Advisor(String firstName, String lastName, InstructorID id, List<Lecture> listOfLectures, Calendar dateOfEntry, List<Student> listOfStudents,
			List<LectureRegistrationApplication> listOfApplications, InstructorType instructorType) {
		super(firstName, lastName, id, listOfLectures, dateOfEntry, instructorType);
		this.listOfStudents = listOfStudents;
		this.listOfApplications = listOfApplications;
		if(this.listOfStudents == null) {
			this.listOfStudents = new ArrayList<Student>();
		}
		if (this.listOfApplications == null) {
			this.listOfApplications = new ArrayList<LectureRegistrationApplication>();
		}
	}

}
